project/Component/NefCommon/Models/RenderingURL.swift

```diff
@@ -9,8 +9,8 @@ public struct RenderingURL {
     public let title: String
 
     public var escapedTitle: String {
-        title.lowercased().replacingOccurrences(of: "?", with: "-")
-                          .replacingOccurrences(of: " ", with: "-")
+        String(title.lowercased().unicodeScalars.filter { !$0.isAlphanumeric || $0 == " " || $0 == "-" })
```

Comment:
Aren't `" "` and `"-"` already checked in the `!$0.isAlphanumeric`?

Reply:
It is a good question, in fact like you, I thought these characters are in the non-base characters, but it isn't

FYI: CharacterSet.alphanumeric contains:
```
A character set containing the characters in Unicode General Categories L*, M*, and N*.
```
and we can check the category M*
1. [Unicode Character Category 'Mark, Spacing Combining](https://www.fileformat.info/info/unicode/category/Mc/index.htm)
2. [Unicode Character Category 'Mark, Enclosing'](https://www.fileformat.info/info/unicode/category/Me/index.htm)
3. [Unicode Character Category 'Mark, Nonspacing'](https://www.fileformat.info/info/unicode/category/Mn/index.htm)

and also, if we check directly to the character set it doesn't return it:
```
CharacterSet.alphanumerics.contains(" ") // false
CharacterSet.alphanumerics.contains("-") // false
```